zanata-war/src/main/java/org/zanata/service/impl/TranslationServiceImpl.java

```diff
@@ -186,6 +190,9 @@
         validateReviewPermissionIfApplicable(translationRequests,
                 projectIteration, hLocale);
 
+        Map<DocumentLocaleKey, List<TextFlowTargetState>> eventMap =
+            Maps.newHashMap();
```

Comment:
I think the code relating to eventMap would be more readable if you were to use a MultiMap.